subsys/bluetooth/mesh/light_hue_srv.c

```diff
@@ -459,10 +459,18 @@ static int hue_srv_init(struct bt_mesh_model *model)
 	srv->emds_entry.entry.id = EMDS_MODEL_ID(model);
 	srv->emds_entry.entry.data = (uint8_t *)&srv->transient;
 	srv->emds_entry.entry.len = sizeof(srv->transient);
-	int err = emds_entry_add(&srv->emds_entry);
+	int err_emds = emds_entry_add(&srv->emds_entry);
```

Comment:
The same variable can be used for error checking. I think you can use `(void)err;` to avoid warning.

Reply:
The problem was that the err related to emds is inside of a #if IS_ENABLED, which means that it might not be declared and thus reusable in the #if statement related to correspondence. I tried moving the err declaration outside of the #ifs, which results in a warning in the case none of the #ifs are true.

Reply:
Not sure if this is an ideal fix to this issue, but what if you change the lats return statement of this function to:

```
err = bt_mesh_model_extend(model, srv->lvl.model);
return err;
```

and declare `int err` at the top of the function? I think that would let you use a single err variable at least.

Reply:
Hmm, the problem is only present in `light_sat_srv.c`, `light_hue_srv.c` and `light_temp_srv.c`, so I think Anders' suggestion looks more clean and should be used in these cases. Thoughts @PavelVPV ?

Reply:
Sounds good

Reply:
Fixed now!